### PR TITLE
nsqd: fix topic.messageCount on error

### PR DESCRIPTION
**Pull request by SwanSpouse**
As the topic says, topic.messageCount may be less than actually when using topic.PutMessages to write message to backend occurs an error.

### CODE REVIEW

File: nsqd/topic.go [hunk 1]
```diff
@@ -199,8 +199,8 @@ func (t *Topic) PutMessages(msgs []*Message) error {
 		if err != nil {
 			return err
 		}
+		atomic.AddUint64(&t.messageCount, 1)
```
Comment (mreiferson) [inline nsqd/topic.go]: instead of single increments, let's maintain a local counter and then perform a single atomic add below on line 203?

**Comment by mreiferson** [comment]
@SwanSpouse thanks for this PR, just a minor comment!

**Comment by ploxiln** [comment]
Yup, good find!

File: nsqd/topic.go (hunk 1 above)
Comment (mreiferson) [reply nsqd/topic.go]: ahh right, I completely missed the fact that it was returning, too. thx 🙏

Comment (SwanSpouse) [reply nsqd/topic.go]: I have modified my codes according to @ploxiln 's advice. Please take a look 🙂

**Pull request merged**